packages/block-library/src/column/edit.js

```diff
@@ -6,52 +6,114 @@ import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { InnerBlocks, BlockControls, BlockVerticalAlignmentToolbar } from '@wordpress/block-editor';
+import {
+	InnerBlocks,
+	BlockControls,
+	BlockVerticalAlignmentToolbar,
+	InspectorControls,
+} from '@wordpress/block-editor';
+import { PanelBody, RangeControl } from '@wordpress/components';
 import { withDispatch, withSelect } from '@wordpress/data';
 import { compose } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
 
-const ColumnEdit = ( { attributes, updateAlignment } ) => {
-	const { verticalAlignment } = attributes;
+function ColumnEdit( {
+	attributes,
+	updateAlignment,
+	updateWidth,
+	hasChildBlocks,
+} ) {
+	const { verticalAlignment, width } = attributes;
 
 	const classes = classnames( 'block-core-columns', {
 		[ `is-vertically-aligned-${ verticalAlignment }` ]: verticalAlignment,
 	} );
 
-	const onChange = ( alignment ) => updateAlignment( alignment );
-
 	return (
 		<div className={ classes }>
 			<BlockControls>
 				<BlockVerticalAlignmentToolbar
-					onChange={ onChange }
+					onChange={ updateAlignment }
 					value={ verticalAlignment }
 				/>
 			</BlockControls>
-			<InnerBlocks templateLock={ false } />
+			<InspectorControls>
+				<PanelBody title={ __( 'Column Settings' ) }>
+					<RangeControl
+						label={ __( 'Width' ) }
+						value={ width }
+						onChange={ updateWidth }
+						min={ 0 }
+						max={ 100 }
+						required
+					/>
+				</PanelBody>
+			</InspectorControls>
+			<InnerBlocks
+				templateLock={ false }
+				renderAppender={ (
+					hasChildBlocks ?
+						undefined :
+						() => <InnerBlocks.ButtonBlockAppender />
+				) }
+			/>
 		</div>
 	);
-};
+}
 
 export default compose(
-	withSelect( ( select, { clientId } ) => {
-		const { getBlockRootClientId } = select( 'core/editor' );
+	withSelect( ( select, ownProps ) => {
+		const { clientId } = ownProps;
+		const { getBlockOrder } = select( 'core/block-editor' );
 
 		return {
-			parentColumnsBlockClientId: getBlockRootClientId( clientId ),
+			hasChildBlocks: getBlockOrder( clientId ).length > 0,
 		};
 	} ),
-	withDispatch( ( dispatch, { clientId, parentColumnsBlockClientId } ) => {
+	withDispatch( ( dispatch, ownProps, registry ) => {
```

Comment:
Nit: Any reason you didn't destructure registry

Reply:
> Nit: Any reason you didn't destructure registry

I'm personally not a fan of destructuring in an arguments signature, since it loses context of _what it is_ that is being destructured. To a lesser degree, I find it hard to visually distinguish between proper arguments, and properties destructured (attention to the `{` and `}`).